File: src/01-easy/mime-type/solution.js
const N = parseInt(readline());
const Q = parseInt(readline());
const mimes = {};
for (let i = 0; i < N; i++) {
    const [ext, mt] = readline().split(' ');
    mimes[ext.toLowerCase()] = mt;
}
for (let i = 0; i < Q; i++) {
    const fname = readline();
    const dotIdx = fname.lastIndexOf('.');
    if (dotIdx === -1 || dotIdx === fname.length - 1) {
        console.log('UNKNOWN');
    } else {
        const ext = fname.substring(dotIdx + 1).toLowerCase();
        console.log(mimes[ext] || 'UNKNOWN');
    }
}

File: src/01-easy/mime-type/solution.clj
(ns Solution
  (:gen-class)
  (:require [clojure.string :as str]))
(defn -main [& args]
  (let [n (Integer/parseInt (read-line))
        q (Integer/parseInt (read-line))
        m (into {} (for [_ (range n)
                         :let [p (str/split (read-line) #" ")]]
                     [(.toLowerCase (first p)) (second p)]))]
    (dotimes [_ q]
      (let [f (read-line)
            dot (.lastIndexOf f ".")]
        (if (= dot -1)
          (println "UNKNOWN")
          (println (get m (.toLowerCase (.substring f (inc dot))) "UNKNOWN")))))))
